cpp-data-aggregation/aggregation_consumer.cpp

```diff
@@ -11,40 +11,59 @@ int get_iterations() {
 void run_aggregation_consumer(std::ofstream& timing_file,
                               int list_length)
 {
+    std::string context("Run Data Aggregation consumer");
+
+    //Initializing rank
     int rank;
     MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+    log_data(context, LLDebug, "rank: " + std::to_string(rank) + " initiated");
 
+    //Indicate Client creation
     if (rank == 0)
+        log_data(context, LLInfo, "Connecting clients");
         std::cout << "Connecting clients" << std::endl;
 
     // Connect the client and save connection time
     double constructor_start = MPI_Wtime();
-    SmartRedis::Client client(true);
+    SmartRedis::Client client(true, context);
     double constructor_end = MPI_Wtime();
     double delta_t = constructor_end - constructor_start;
     timing_file << rank << "," << "client()" << ","
                 << delta_t << "\n";
+    //print rank # storing client() for debugging
+    std::string client_text = "client() time stored for rank: ";
+    client_text += std::to_string(rank);
```

Comment:
you can combine these two lines:
`std::string client_text = "client() time stored for rank: " + std::to_string(rank);`

Reply:
thank u for this helpful advice I did not realize